### PR TITLE
Update poetry version to latest

### PR DESCRIPTION
**Pull request by tanertopal**
### Description

Updates poetry to the latest version. There will be a subsequent PR for baselines. Some of them don't seem to correctly install right now.

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.dev/docs/writing-documentation.html)
- [ ] Update the changelog entry below
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.dev/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: baselines/baseline_template/pyproject.toml [hunk 1]
```diff
@@ -44,7 +44,7 @@ hydra-core = "1.3.2" # don't change this
 [tool.poetry.dev-dependencies]
 isort = "==5.11.5"
 black = "==23.1.0"
-docformatter = "==1.5.1"
+docformatter = "==1.7.1"
```
Comment (danieljanes) [inline baselines/baseline_template/pyproject.toml:47]: Is this intentional?

Comment (danieljanes) [reply baselines/baseline_template/pyproject.toml:47]: The latest version of docformatter is `1.7.5`

Comment (tanertopal) [reply baselines/baseline_template/pyproject.toml:47]: No, it wasn't. I'll remove that.

**Comment by tanertopal** [comment]
Moving back to draft until its green

**Review by danieljanes — APPROVED** [review]

**Pull request merged**